pkg/cmd/pipeline/start.go

```diff
@@ -97,49 +100,145 @@ tkn pipeline start foo --param NAME=VALUE --resource source=scaffold-git  -s Ser
 	c.Flags().StringSliceVarP(&res, "resource", "r", []string{}, "pass the resource name and ref")
 	c.Flags().StringSliceVarP(&params, "param", "p", []string{}, "pass the param")
 	c.Flags().StringVarP(&svc, "serviceaccount", "s", svc, "pass the serviceaccount name")
-
+	c.Flags().BoolVarP(&last, "last", "l", false, "re-run the pipeline using last pipelinerun values")
 	return c
 }
 
 func startPipeline(out io.Writer, opt startOptions, p cli.Params, pName string) error {
+	pr := &v1alpha1.PipelineRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:    p.Namespace(),
+			GenerateName: pName + "-run-",
+		},
+		Spec: v1alpha1.PipelineRunSpec{
+			PipelineRef: v1alpha1.PipelineRef{Name: pName},
+		},
+	}
+
+	if opt.Last {
+		prLast, err := lastPipelineRun(p, pName)
+		if err != nil {
+			return err
+		}
+		pr.Spec.Resources = prLast.Spec.Resources
+		pr.Spec.Params = prLast.Spec.Params
+		pr.Spec.ServiceAccount = prLast.Spec.ServiceAccount
+	}
+
+	if err := mergeRes(pr, opt.Resources); err != nil {
+		return err
+	}
+
+	if err := mergeParam(pr, opt.Params); err != nil {
+		return err
+	}
+
+	if len(opt.ServiceAccountName) > 0 {
+		pr.Spec.ServiceAccount = opt.ServiceAccountName
+	}
 
 	cs, err := p.Clients()
 	if err != nil {
 		return err
 	}
 
-	res, err := parseRes(opt.Resources)
+	pr, err = cs.Tekton.TektonV1alpha1().PipelineRuns(p.Namespace()).Create(pr)
 	if err != nil {
 		return err
 	}
 
-	params, err := parseParam(opt.Params)
+	fmt.Fprintf(out, "Pipelinerun started: %s\n", pr.Name)
+	return nil
+}
+
+func mergeRes(pr *v1alpha1.PipelineRun, optRes []string) error {
+	res, err := parseRes(optRes)
 	if err != nil {
 		return err
 	}
 
-	pr := &v1alpha1.PipelineRun{
-		ObjectMeta: metav1.ObjectMeta{
-			Namespace:    p.Namespace(),
-			GenerateName: pName + "-run-",
-		},
-		Spec: v1alpha1.PipelineRunSpec{
-			PipelineRef:    v1alpha1.PipelineRef{Name: pName},
-			ServiceAccount: opt.ServiceAccountName,
-			Resources:      res,
-			Params:         params,
-		},
+	if len(res) == 0 {
+		return nil
 	}
 
-	pr, err = cs.Tekton.TektonV1alpha1().PipelineRuns(p.Namespace()).Create(pr)
+	for i := range res {
```

Comment:
Any reason for using a loop in a loop instead of a `map[string]int` to store the "overriden" resoruce (and param below) and the corresponding index ?

Reply:
Because I assumed that extra params or resources might be passed. It wasn't making sense to use map that way. I changed my approach to insert params or res to map now instead of array. Now I am using map.